test/test_rtfcre_dict.py

```diff
@@ -210,6 +217,49 @@ def test_format_translation(self):
             )
             self.assertEqual(result, expected, msg=msg)
 
+    def test_round_trip_json(self):
+        styles = {1: 'Normal', 2: 'Continuation'}
+
+        convert = TranslationConverter(styles)
+
+        err = (
+            'Expected {initial} to turn into {end} in round trip.\n' +
+            'Instead, {result} was found. See process:\n{states}'
+        )
+
+        cases = (
+            ('{^\n^}',),
+            ('{#Return}',),
+            ('{^^}', '{^}{^}'),
+            ('{^test}',),
+            ('{^toast^}',),
+            ('Mr. {-|}',),
+            ('Mr.{-|}',),
+            ('state your{^ ^} name, please{^\n\n^}{-|}',),
+            ('{PLOVER:ON}',),
+            ('{^~|\n\n^}',),
+            ('{^~|\n\n}',),
+        )
+        for case in cases:
+            initial = case[0]
+            expected = case[-1]
+            states = [initial]
+            settled = False
+            iterations = 0
+            while not settled and iterations < 10:
+                states.append(format_translation(states[-1]))
+                states.append(convert(states[-1]))
+                if states[-1] == states[-3]:
+                    settled = True
+                iterations += 1
+
+            self.assertEqual(expected, states[-1],
+                             err.format(initial=repr(initial),
+                                        end=repr(expected),
+                                        result=repr(states[-1]),
+                                        states=states))
+
```

Comment:
I would prefer something simpler:
```python
    def test_round_trip_json(self):
        styles = {1: 'Normal', 2: 'Continuation'}

        convert = TranslationConverter(styles)

        err = (
            'Expected {initial} to turn into one of {expected} in round trip.\n' +
            'Instead, {result} was found. See process:\n{states}'
        )

        cases = (
            ('{^\n^}',),
            ('{#Return}',),
            ('{^^}', '{^}{^}'),
            ('{^test}',),
            ('{^toast^}',),
            ('Mr. {-|}',),
            ('Mr.{-|}',),
            ('state your{^ ^} name, please{^\n\n^}{-|}',),
            ('{PLOVER:ON}',),
            ('{^~|\n\n^}',),
            ('{^~|\n\n}',),
        )
        for case in cases:
            for translation in case:
                states = [translation]
                states.append(format_translation(states[-1]))
                states.append(convert(states[-1]))
                self.assertIn(states[-1], case,
                              err.format(initial=repr(states[0]),
                                         expected=repr(case),
                                         result=repr(states[-1]),
                                         states=states))
```